velox/exec/ExchangeClient.cpp

```diff
@@ -21,7 +21,7 @@ void ExchangeClient::addRemoteTaskId(const std::string& taskId) {
   RequestSpec requestSpec;
   std::shared_ptr<ExchangeSource> toClose;
   {
-    std::lock_guard<std::mutex> l(queue_->mutex());
+    std::lock_guard<std::mutex> l(mutex_);
```

Comment:
There is a call to queue_->addSourceLocked() which requires lock over queue's mutex. Please, double check.